packages/core/lib/metrics/MessageProcessingTimePrometheusMetric.ts

```diff
@@ -0,0 +1,84 @@
+import type { MessageMetricsManager, ProcessedMessageMetadata } from '@message-queue-toolkit/core'
+import type { Histogram } from 'prom-client'
+import promClient from 'prom-client'
+
+/**
+ * Parameters used for registering message processing metrics in Prometheus
+ */
+type PrometheusMetricParams<T extends object> = {
+  /**
+   * Prometheus metric name
+   */
+  name: string
+
+  /**
+   * Prometheus metric description
+   */
+  helpDescription: string
+
+  /**
+   * Buckets used to configure Histogram metric
+   */
+  buckets: number[]
+
+  /**
+   * Message version used as a label - can be static string or method resolving version based on payload
+   */
+  messageVersion?: string | ((messageMetadata: ProcessedMessageMetadata<T>) => string | undefined)
```

Comment:
Message version can be extracted in multiple ways depending on message schema, so allowing to provide it as a parameter